### PR TITLE
Data-Driven Keyboard Conversions: B

### PR DESCRIPTION
**Pull request by noroadsleft**
## Description

Converts configuration definitions and rules to data-driven where applicable. Renames `info.json` to `keyboard.json` in instances where `rules.mk` can be safely deleted.

- keyboards with names beginning with `b`

:warning: **NOTE** :warning: : I've deliberately skipped editing anything in `keyboards/bastardkb/` because there's so many nested variants that I couldn't be bothered. I'll come back around for those.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/barleycorn_smd/info.json [hunk 1]
```diff
@@ -33,6 +45,11 @@
     },
     "processor": "atmega32u4",
     "bootloader": "atmel-dfu",
+    "matrix_pins": {
+        "cols": ["D4", "D6", "D7", "B4", "B5", "B6", "C6", "C7", "D5", "D5", "D5", "D5", "D5", "D5", "D5", "D5", "D5", "D5"],
+        "rows": ["F5", "F4", "F1", "F0", "F6"]
+    },
+    "diode_direction": "COL2ROW",
```
Comment (fauxpark) [inline keyboards/barleycorn_smd/info.json:52]: As above.

File: keyboards/bandominedoni/bandominedoni.c [hunk 2]
```diff
@@ -16,58 +16,6 @@
 
 #include "bandominedoni.h"
 
-#ifdef RGB_MATRIX_ENABLE
```
Comment (fauxpark) [inline keyboards/bandominedoni/bandominedoni.c:19]: Do these migrations produce the same output?

**Review by fauxpark — CHANGES_REQUESTED** [review]
One last note: NoScript really seems to disagree with the sheer size of these PRs making it painfully laggy (weird, it's usually RGH trying to do all the whitespace chars). Extracting the locking migrations to separate PRs would probably be faster for me to review.

File: keyboards/bandominedoni/bandominedoni.c (hunk 2 above)
Comment (noroadsleft) [reply keyboards/bandominedoni/bandominedoni.c:19]: It did the previous time I checked it, but just now it didn't. :man_shrugging: Reverting.

File: keyboards/basekeys/trifecta/keyboard.json [hunk 3]
```diff
@@ -102,7 +108,89 @@
             "multisplash": true,
             "solid_splash": true,
             "solid_multisplash": true
-        }
+        },
```
Comment (fauxpark) [inline keyboards/basekeys/trifecta/keyboard.json:111]: Needs `"driver": "ws2812"` above; RGB Matrix is not enabled by default so CI didn't catch this.

**Comment by finrod09** [comment]
re `bastardkb`: A removal of legacy variants and flattening the directory structure is planned (I have already begun work but need some more time and internal review); We should speak about this to avoid duplication of efforts.

**Review by fauxpark — APPROVED** [review]

**Pull request merged**